pkg/apis/infinispan/v1/infinispan_types.go

```diff
@@ -91,11 +91,10 @@ type InfinispanSitesLocalSpec struct {
 
 type InfinispanSiteLocationSpec struct {
 	Name        string `json:"name"`
+	// +kubebuilder:validation:Pattern=`(^(kubernetes|minikube|openshift):\/\/(([a-z0-9]|[a-z0-9][a-z0-9\-]*[a-z0-9])\.)*([a-z0-9]|[a-z0-9][a-z0-9\-]*[a-z0-9])*(:[0-9]+)+$)|(^(xsite):\/\/(([a-z0-9]|[a-z0-9][a-z0-9\-]*[a-z0-9])\.)*([a-z0-9]|[a-z0-9][a-z0-9\-]*[a-z0-9])*(:[0-9]+)*$)`
+	URL         string `json:"url"`
 	Namespace   string `json:"namespace,optional,omitempty"`
 	ClusterName string `json:"clusterName,optional,omitempty"`
-	Host        string `json:"host,optional,omitempty"`
```

Comment:
This also breaks backward compatibility. I would consider a transition phase: i.e. keep these fields and, if empty populate URL field with them.
wdyt?

Reply:
Probably yes, but arguments to change this immediately are:
- Not yet productized in downstream
- Less use case for this options
- Less than 2 month in release on OperatorHub
- Brakes original design doc v3

Reply:
Was the release on OperatorHub marked as experimental/dev?

Reply:
@ryanemerson we do not update stable and development channels for a long time.Latest stable version is 1.1.2 and development-preview is 2.0.4. This means that 2.1.x is not stable and can be changed.

Reply:
No we don't have that distinction since we moved to x.y.z channel names.

We don't need to document the old use case any more, just provide support to avoid problems with upgrade

Reply:
Is there anyway on operatorhub to mark a stream as being `dev or experimental`? Even adding a README would suffice if there's nothing else.

This would make it so that the compatiblity guarantees with provide for users are explicit. In the same way that our use of `DevXX` tags vs `.Final` is.

> This means that 2.1.x is not stable and can be changed.

Where have we informed users of this?

Reply:
> Where have we informed users of this?

It seems to no information and we don't support named channels.
At the end, we need some rules what we can brake in the changes and what not. Probably short release and not downstream can be changed

Reply:
> At the end, we need some rules what we can brake in the changes and what not.

Totally agree. We need this flexibility for productive development, however we must communicate this with upstream users some how.


> Probably short release and not downstream can be changed

I'm not sure what you mean by "short release".

Channels which are used downstream have to have backwards compatibility like you say. But so does any upstream channel that is beyond a `x-y-0.Final` tag. 

Where I think we have failed with the `2.1.x` tag, is on operatorhub etc we have not documented to the user when the dev cycle ended and when the stable version of that channel started. It appears as if it's one continous stream of releases, which will cause confusion to a user if OLM upgrades their clusters now and it breaks due to incompatible CR spec changes.

I'm not sure why the "dev-preview" channel was abandoned as I wasn't focusing on the operator then. Why can't we always use the "dev-preview" channel for `master` releases and then only release to the `x.y.z` channels after the first `.Final` tag once things are stable?

Reply:
> I'm not sure what you mean by "short release".

Sorry, minor of course.
Yes, agree with you. I think the serious brakes are not possible between `x.y.z.Final` releases

Reply:
We really shouldn't remove anything from V1 of the CRD. It's supposed to be stable. Operator channel/CSV version isn't only thing that's ensuring in the stack providing availability of a particular feature. (If you touch CRD you mess up with the users of the older versions. If you touch the code only, the CRD will imply that the functionality is there).

Reply:
Yeah, @Crumby totally agreed. But this changes was introduced in intermediate release and not stable. Not included in downstream also. We should prevent to make changes like this, but missing on review stage comparing with design document

Reply:
As discussed on Zulip, we should initially keep port+url and transform the CR when users use this field in the next upstream release. The `host+port` fields will then be removed on the subsequent upstream release to ensure that it does not affect downstream.

Reply:
Yes, sure, @ryanemerson 
Kept host and port fileds and add URL generator on upgrade step